From dea9c9ce6fef2eccd3b8cd55e30e7af95a7e9a0b Mon Sep 17 00:00:00 2001
From: Sridharan S N <quic_sridsn@quicinc.com>
Date: Tue, 21 Mar 2023 15:56:27 +0530
Subject: [PATCH 176/281] dt-bindings: phy: qcom,qmp-pcie: Add ipq9574
 compatible

Add the compatible for the PCIe QMP PHYs found on IPQ9574

Change-Id: I000c2a325c26099bff10d9470483987cf3d54c96
Signed-off-by: Devi Priya <quic_devipriy@quicinc.com>
Signed-off-by: Sridharan S N <quic_sridsn@quicinc.com>
---
 .../bindings/phy/qcom,qmp-pcie-phy.yaml       | 28 +++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/Documentation/devicetree/bindings/phy/qcom,qmp-pcie-phy.yaml b/Documentation/devicetree/bindings/phy/qcom,qmp-pcie-phy.yaml
index 324ad7d03a38..578a34b91a4b 100644
--- a/Documentation/devicetree/bindings/phy/qcom,qmp-pcie-phy.yaml
+++ b/Documentation/devicetree/bindings/phy/qcom,qmp-pcie-phy.yaml
@@ -19,6 +19,8 @@ properties:
       - qcom,ipq6018-qmp-pcie-phy
       - qcom,ipq8074-qmp-gen3-pcie-phy
       - qcom,ipq8074-qmp-pcie-phy
+      - qcom,ipq9574-qmp-gen3x1-pcie-phy
+      - qcom,ipq9574-qmp-gen3x2-pcie-phy
       - qcom,msm8998-qmp-pcie-phy
       - qcom,sc8180x-qmp-pcie-phy
       - qcom,sdm845-qhp-pcie-phy
@@ -162,6 +164,30 @@ allOf:
             - const: phy
             - const: common
 
+  - if:
+      properties:
+        compatible:
+          contains:
+            enum:
+              - qcom,ipq9574-qmp-gen3x1-pcie-phy
+              - qcom,ipq9574-qmp-gen3x2-pcie-phy
+    then:
+      properties:
+        clocks:
+          maxItems: 4
+        clock-names:
+          items:
+            - const: aux
+            - const: cfg_ahb
+            - const: anoc_lane
+            - const: snoc_lane
+        resets:
+          maxItems: 2
+        reset-names:
+          items:
+            - const: phy
+            - const: common
+
   - if:
       properties:
         compatible:
@@ -200,6 +226,7 @@ allOf:
         compatible:
           contains:
             enum:
+              - qcom,ipq9574-qmp-gen3x2-pcie-phy
               - qcom,sm8250-qmp-gen3x2-pcie-phy
               - qcom,sm8250-qmp-modem-pcie-phy
               - qcom,sm8450-qmp-gen4x2-pcie-phy
@@ -221,6 +248,7 @@ allOf:
         compatible:
           contains:
             enum:
+              - qcom,ipq9574-qmp-gen3x1-pcie-phy
               - qcom,sc8180x-qmp-pcie-phy
               - qcom,sdm845-qmp-pcie-phy
               - qcom,sdx55-qmp-pcie-phy
-- 
2.17.1

